src/gui/editorwidgets/qgsvaluemapwidgetwrapper.cpp

```diff
@@ -78,5 +78,11 @@ void QgsValueMapWidgetWrapper::updateValues( const QVariant &value, const QVaria
     v = value.toString();
 
   if ( mComboBox )
+  {
+    if ( mComboBox->findData( v ) == -1 && !( v.startsWith( '(' ) && v.endsWith( ')' ) ) )
```

Comment:
@elpaso 
Looking at your latest ValueMap pr I stumbled upon this.
What is this check for?

If I'm reading this correctly, `v` refers to the *stored* value / userData, while parentheses are added to the *represented* value.

If I'm reading this correctly, an unmatched entry `(none)` will currently not show up in the list. I think it should show up as `((none))` (or similar).

Reply:
IIRC this was added to avoid double addition of `()`, and yes, I know it's not robust.

Reply:
This code smells.
It's probably covering up an issue somewhere else where value/key are mixed up.

Reply:
I agree ... I'll see if I can reproduce the double `()` issue without that check.

Reply:
Thanks !

Reply:
https://github.com/qgis/QGIS/pull/37047